virtwho/virt/xen/xen.py

```diff
@@ -187,8 +187,9 @@ def _run(self):
                 events = []
 
             if initial or len(events) > 0 or delta > 0:
-                assoc = self.getHostGuestMapping()
-                self._send_data(virt.HostGuestAssociationReport(self.config, assoc))
+                if self.is_registered() is True:
```

Comment:
I'm not entirely certain that this will solve the issue. I'm guessing the interrupted sys call happens due to this thread being killed while active waiting.